### PR TITLE
OSCAR: init at 1.2.0

### PR DESCRIPTION
**Pull request by roconnor**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/OSCAR/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    sha256 = "10r37d8c2avr167n2s9lhld1c9hmckm444fq163z1jsy9jpid6mg";
+  };
+  buildInputs = [ qtbase qttools qtserialport libGLU ];
+  nativeBuildInputs = [ qmake ];
+  postPatch = ''
+    substituteInPlace oscar/oscar.pro --replace "/bin/bash" "${stdenv.shell}"
+  '';
+  qmakeFlags = [ "OSCAR_QT.pro" ];
+
+# help browser was removed 'temporarily' in https://gitlab.com/pholy/OSCAR-code/-/commit/57c3e4c33ccdd2d0eddedbc24c0e4f2969da3841
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/OSCAR/default.nix:19]: Can you move this next to the commented line?

File: pkgs/applications/misc/OSCAR/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  installPhase = ''
+    install -d $out/bin
+    install -d $out/share/OSCAR/Help
+    install -d $out/share/OSCAR/Html
+    install -d $out/share/OSCAR/Translations
+    install -d $out/share/icons/OSCAR
+    install -d $out/share/applications
+    install -T oscar/OSCAR $out/bin/OSCAR
+#    install oscar/Help/* $out/share/OSCAR/Help
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/OSCAR/default.nix:28]:
```suggestion
    # install oscar/Help/* $out/share/OSCAR/Help
```

File: pkgs/applications/misc/OSCAR/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    repo = "OSCAR-code";
+    rev = "v${version}";
+    sha256 = "10r37d8c2avr167n2s9lhld1c9hmckm444fq163z1jsy9jpid6mg";
+  };
+  buildInputs = [ qtbase qttools qtserialport libGLU ];
+  nativeBuildInputs = [ qmake ];
+  postPatch = ''
+    substituteInPlace oscar/oscar.pro --replace "/bin/bash" "${stdenv.shell}"
+  '';
+  qmakeFlags = [ "OSCAR_QT.pro" ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/OSCAR/default.nix:17]:
```suggestion
  '';

  qmakeFlags = [ "OSCAR_QT.pro" ];
```

File: pkgs/applications/misc/OSCAR/default.nix [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+ in
+mkDerivation {
+  name = "OSCAR-${version}";
+  src = fetchFromGitLab {
+    owner = "pholy";
+    repo = "OSCAR-code";
+    rev = "v${version}";
+    sha256 = "10r37d8c2avr167n2s9lhld1c9hmckm444fq163z1jsy9jpid6mg";
+  };
+  buildInputs = [ qtbase qttools qtserialport libGLU ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/OSCAR/default.nix:12]:
```suggestion
  };

  buildInputs = [ qtbase qttools qtserialport libGLU ];
```

File: pkgs/applications/misc/OSCAR/default.nix [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  nativeBuildInputs = [ qmake ];
+  postPatch = ''
+    substituteInPlace oscar/oscar.pro --replace "/bin/bash" "${stdenv.shell}"
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/OSCAR/default.nix:15]: You can probably use patchShebangs here.

File: pkgs/applications/misc/OSCAR/default.nix [hunk 6]
```diff
@@ -0,0 +1,41 @@
+{ stdenv, mkDerivation, fetchFromGitLab, qmake, qtbase, qttools, qtserialport, libGLU }:
+let version = "1.2.0";
+ in
+mkDerivation {
+  name = "OSCAR-${version}";
+  src = fetchFromGitLab {
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/OSCAR/default.nix:6]:
```suggestion
  name = "OSCAR-${version}";

  src = fetchFromGitLab {
```

File: pkgs/applications/misc/OSCAR/default.nix [hunk 7]
```diff
@@ -0,0 +1,41 @@
+{ stdenv, mkDerivation, fetchFromGitLab, qmake, qtbase, qttools, qtserialport, libGLU }:
+let version = "1.2.0";
+ in
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/OSCAR/default.nix:3]: Can we move this into the derivation?

**Review by SuperSandro2000 — APPROVED** [review]

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Pressed the wrong button.

File: pkgs/applications/misc/OSCAR/default.nix [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+mkDerivation {
+  name = "OSCAR-${version}";
```
Comment (AndersonTorres) [inline pkgs/applications/misc/OSCAR/default.nix:5]: pname and version

File: pkgs/applications/misc/OSCAR/default.nix (hunk 5 above)
Comment (roconnor) [reply pkgs/applications/misc/OSCAR/default.nix:15]: In this particular case, it isn't a shebang: https://gitlab.com/pholy/OSCAR-code/-/blob/master/oscar/oscar.pro#L76-78

File: pkgs/applications/misc/OSCAR/default.nix (hunk 1 above)
Comment (roconnor) [reply pkgs/applications/misc/OSCAR/default.nix]: Done.

File: pkgs/applications/misc/OSCAR/default.nix (hunk 2 above)
Comment (roconnor) [reply pkgs/applications/misc/OSCAR/default.nix]: Done.

File: pkgs/applications/misc/OSCAR/default.nix (hunk 3 above)
Comment (roconnor) [reply pkgs/applications/misc/OSCAR/default.nix]: Done.

File: pkgs/applications/misc/OSCAR/default.nix (hunk 4 above)
Comment (roconnor) [reply pkgs/applications/misc/OSCAR/default.nix]: Done.

File: pkgs/applications/misc/OSCAR/default.nix (hunk 6 above)
Comment (roconnor) [reply pkgs/applications/misc/OSCAR/default.nix]: Done.

File: pkgs/applications/misc/OSCAR/default.nix (hunk 7 above)
Comment (roconnor) [reply pkgs/applications/misc/OSCAR/default.nix]: Done.

File: pkgs/applications/misc/OSCAR/default.nix (hunk 8 above)
Comment (roconnor) [reply pkgs/applications/misc/OSCAR/default.nix]: Done.

**Review by SuperSandro2000 — COMMENTED** [review]
When building on darwin I get:

```
clang++ -stdlib=libc++ -o OSCAR.app/Contents/MacOS/OSCAR checkupdates.o common_gui.o cprogressbar.o daily.o exportcsv.o main.o mainwindow.o newprofile.o overview.o preferencesdialog.o reports.o sessionbar.o version.o gFlagsLine.o gFooBar.o gGraph.o gGraphView.o glcommon.o gLineChart.o gLineOverlay.o gSegmentChart.o gspacer.o gStatsLine.o gSummaryChart.o gXAxis.o gYAxis.o layer.o calcs.o common.o day.o event.o machine.o machine_loader.o preferences.o profiles.o schema.o session.o cms50_loader.o dreem_loader.o icon_loader.o intellipap_loader.o mseries_loader.o prs1_loader.o resmed_loader.o resmed_EDFinfo.o somnopose_loader.o viatom_loader.o zeo_loader.o zip.o miniz.o csv.o translation.o statistics.o oximeterimport.o deviceconnection.o xmlreplay.o serialoximeter.o md300w1_loader.o gSessionTimesChart.o gPressureChart.o logger.o machine_common.o weinmann_loader.o gdailysummary.o MinutesAtPressure.o journal.o progressdialog.o cms50f37_loader.o profileselector.o appsettings.o edfparser.o aboutdialog.o welcome.o qrc_Resources.o moc_checkupdates.o moc_daily.o moc_exportcsv.o moc_mainwindow.o moc_newprofile.o moc_overview.o moc_preferencesdialog.o moc_sessionbar.o moc_gGraph.o moc_gGraphView.o moc_machine_loader.o moc_cms50_loader.o moc_icon_loader.o moc_prs1_loader.o moc_resmed_loader.o moc_zip.o moc_statistics.o moc_oximeterimport.o moc_deviceconnection.o moc_serialoximeter.o moc_md300w1_loader.o moc_logger.o moc_progressdialog.o moc_cms50f37_loader.o moc_profileselector.o moc_aboutdialog.o moc_welcome.o moc_mytextbrowser.o   -lz /nix/store/nsk1zf0vf7mzz68may1vcw8jrvgpc4y9-qtbase-5.12.7/lib/libQt5PrintSupport.dylib /nix/store/nsk1zf0vf7mzz68may1vcw8jrvgpc4y9-qtbase-5.12.7/lib/libQt5OpenGL.dylib /nix/store/nsk1zf0vf7mzz68may1vcw8jrvgpc4y9-qtbase-5.12.7/lib/libQt5Widgets.dylib /nix/store/nsk1zf0vf7mzz68may1vcw8jrvgpc4y9-qtbase-5.12.7/lib/libQt5Gui.dylib /nix/store/nsk1zf0vf7mzz68may1vcw8jrvgpc4y9-qtbase-5.12.7/lib/libQt5Network.dylib /nix/store/nsk1zf0vf7mzz68may1vcw8jrvgpc4y9-qtbase-5.12.7/lib/libQt5Xml.dylib /nix/store/23c1aapqj8dmjav276c7bvj1s29i9zac-qtserialport-5.12.7/lib/libQt5SerialPort.dylib /nix/store/nsk1zf0vf7mzz68may1vcw8jrvgpc4y9-qtbase-5.12.7/lib/libQt5Core.dylib -framework OpenGL -framework AGL
make[1]: Leaving directory '/private/var/folders/r4/k85s52jx3555ckj0wty615s00000gp/T/nix-build-OSCAR-1.2.0.drv-0/source/oscar'
installing
install: cannot stat 'oscar/OSCAR': No such file or directory
builder for '/nix/store/qx32zggysfgp2sz5nr3680f7pfb5400l-OSCAR-1.2.0.drv' failed with exit code 1
error: build of '/nix/store/qx32zggysfgp2sz5nr3680f7pfb5400l-OSCAR-1.2.0.drv' failed
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101945` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>OSCAR</li>
  </ul>
</details>

**Comment by roconnor** [comment]
I don't know how to install on darwin.

**Comment by roconnor** [comment]
nixpkgs-review pr 101945 works for me:

```
$ nix-store -q --deriver /nix/store/6dn4vrwazddcd12894a8fnv256wyg3h1-OSCAR-1.2.0/bin/OSCAR
/nix/store/20ggyh6f5nm6c28cv5rjafvcknnk3d9w-OSCAR-1.2.0.drv
```

**Comment by roconnor** [comment]
Added `meta.broken` for non-linux.

**Comment by SuperSandro2000** [comment]
> I don't know how to install on darwin.

That is okay. If anyone wants to fix it for darwin they can do it in the future.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101945` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>OSCAR</li>
  </ul>
</details>

**Pull request merged**